duktape/src/main/jni/DuktapeContext.cpp

```diff
@@ -219,3 +228,52 @@ void DuktapeContext::bindInstance(JNIEnv *env, jstring name, jobject object, job
   // Pop the Duktape global object off the stack.
   duk_pop(m_context);
 }
+
+jobject DuktapeContext::call(JNIEnv *env, jstring name, jobject method, jobjectArray args) {
+  duk_push_global_object(m_context);
+  const JString instanceName(env, name);
+  if (!duk_get_prop_string(m_context, -1, instanceName)) {
+    duk_pop(m_context);
+    const jclass illegalArgumentException = env->FindClass("java/lang/IllegalArgumentException");
+    const std::string message = "A global object called " + instanceName.str() + " was not found";
+    env->ThrowNew(illegalArgumentException, message.c_str());
+    return nullptr;
+  }
+
+  // TODO: cache the call wrapper.
```

Comment:
And I guess there’s a similar, weirder problem with the method references also:

1. JavaScript:
    ```
    var x = {
      greet: function() { return "hello"; }
    };
    ```
2. Java:
    ```
    Greeter greeter = duktape.proxy("x", Greeter.class);
    ```
3. JavaScript
    ```
    x.greet = function() { return "howdy"; };
    ```